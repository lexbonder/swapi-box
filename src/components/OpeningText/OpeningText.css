.fade {
  position: relative;
  width: 100%;
  height: 200px;
}

.OpeningText {
  display: flex;
  justify-content: center;
  position: relative;
  height: 200px;
  color: #feda4a;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 200%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  perspective: 400px;
  text-align: justify;
  overflow: hidden;
  border-bottom: 2px solid #feda4a;

}

.crawl {
  position: relative;
  top: 99999px;
  transform-origin: 50% 100%;
  animation: crawl 60s linear;
}

.crawl > .title {
  font-size: 90%;
  text-align: center;
}

.crawl > .title h1 {
  margin: 0 0 100px;
  text-transform: uppercase;
}

@keyframes crawl {
    0% {
    top: 220px;
    transform: rotateX(20deg)  translateZ(0);
  }
    100% { 
    top: -3000px;
    transform: rotateX(30deg) translateZ(-1500px);
  }
}