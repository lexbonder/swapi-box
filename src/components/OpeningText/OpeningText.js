import React from 'react'

const OpeningText = ({title, episode, text}) => {
  return(
    <div>
      <h2>Episode {episode}</h2>
      <h1>{title}</h1>
      <p>{text}</p>
    </div>
  )
}

export default OpeningText;