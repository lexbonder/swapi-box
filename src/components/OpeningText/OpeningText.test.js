import React from 'react';
import {shallow} from 'enzyme';
import OpeningText from './OpeningText'

describe('OpeningText', () => {
  it('should match the snapshot', () => {
    const mockMovieData = {
      openingText: {
        title: 'Return of the Jedi',
        eipsode: '4',
        crawl: 'The super long intro...',
        year: '04/15/1990'
        }}

    const wrapper = shallow(<OpeningText openingText={mockMovieData}/>)

    expect(wrapper).toMatchSnapshot()
  })
})