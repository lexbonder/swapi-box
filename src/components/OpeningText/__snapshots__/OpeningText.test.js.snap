// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpeningText should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OpeningText
    openingText={
        Object {
            "openingText": Object {
              "crawl": "The super long intro...",
              "eipsode": "4",
              "title": "Return of the Jedi",
              "year": "04/15/1990",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Episode 
</h1>,
        <h2 />,
        <p>
           - 
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Episode ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Episode ",
          undefined,
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            " - ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          " - ",
          undefined,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Episode 
</h1>,
          <h2 />,
          <p>
             - 
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Episode ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Episode ",
            undefined,
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              " - ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            " - ",
            undefined,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
